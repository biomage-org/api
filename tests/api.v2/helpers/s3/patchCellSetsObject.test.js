// @ts-nocheck
const patchCellSetsObject = require('../../../../src/api.v2/helpers/s3/patchCellSetsObject');
const getObject = require('../../../../src/api.v2/helpers/s3/getObject');
const putObject = require('../../../../src/api.v2/helpers/s3/putObject');

jest.mock('../../../../src/api.v2/helpers/s3/getObject');
jest.mock('../../../../src/api.v2/helpers/s3/putObject');

const mockExperimentId = 'mock-experiment-id';

const mockCellSets = {
  cellSets: [
    {
      key: 'louvain',
      name: 'louvain clusters',
      rootNode: true,
      type: 'cellSets',
      children: [
        {
          key: 'louvain-0',
          name: 'Cluster 0',
          rootNode: false,
          type: 'cellSets',
          color: '#77aadd',
          cellIds: [0, 1, 2, 3],
        },
      ],
    },
  ],
};

const mockPatch = [
  {
    $match: {
      query: '$[?(@.key == "scratchpad")]',
      value: {
        children: [
          {
            $insert:
              {
                index: '-',
                value:
                  {
                    key: 'new-cluster-1',
                    name: 'New Cluster 1',
                    color: '#3957ff',
                    type: 'cellSets',
                    cellIds: [4, 5, 6],
                  },
              },
          },
        ],
      },
    },
  },
];

describe('patchCellSetsObject', () => {
  it('Works correctly', async () => {
    getObject.mockReturnValueOnce(mockCellSets);

    const result = await patchCellSetsObject(mockExperimentId, mockPatch);

    // Put a modified object
    const putParams = putObject.mock.calls[0][0];

    expect(putParams).toMatchSnapshot();

    // Does not return anything on success
    expect(result).toBeUndefined();
  });
});
