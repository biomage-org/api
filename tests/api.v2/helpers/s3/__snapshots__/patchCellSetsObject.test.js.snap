// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patchCellSetsObject Works correctly 1`] = `
Object {
  "Body": "{\\"cellSets\\":[{\\"key\\":\\"louvain\\",\\"name\\":\\"louvain clusters\\",\\"rootNode\\":true,\\"type\\":\\"cellSets\\",\\"children\\":[{\\"key\\":\\"louvain-0\\",\\"name\\":\\"Cluster 0\\",\\"rootNode\\":false,\\"type\\":\\"cellSets\\",\\"color\\":\\"#77aadd\\",\\"cellIds\\":[0,1,2,3]}]}]}",
  "Bucket": "cell-sets-test",
  "Key": "mock-experiment-id",
}
`;
